src/Java.Interop.Tools.Maven/DefaultPomResolver.cs

```diff
@@ -0,0 +1,23 @@
+using System;
+using System.Collections.Generic;
+using Java.Interop.Tools.Maven.Models;
+
+namespace Java.Interop.Tools.Maven;
+
+public class DefaultPomResolver : IPomResolver
+{
+	readonly Dictionary<string, Project> poms = new ();
+
+	public void Register (Project project)
+	{
+		poms.Add (project.ToString (), project);
+	}
+
+	public virtual Project ResolveRawProject (Artifact artifact)
+	{
+		if (poms.TryGetValue (artifact.ToString (), out var project))
```

Comment:
It feels "implicitly weird" that we require that `Project.ToString()` be identical to `Artifact.ToString()`.  While `Artifact.ToString()` *does* have a comment stating "This is a "well-known" format we use, it should not be changed", `Project.ToString()` *doesn't* (currently), nor is there a comment stating that these formats should be consistent with each other.

I think it would be clearer and more explicit if we instead had "something" that was identically named between the two types to name the association.  Possible ideas:

  * `ArtifactString` ("rhymes" with `ResolvedDependency.ToArtifactString()`)
  * `Group_Artifact_Version` (that's what it contains…)

Reply:
I went with `ArtifactString` and `VersionedArtifactString` throughout the codebase to make this more explicit and consistent.